rediscontroller/utils.py

```diff
@@ -24,3 +38,68 @@ def get_install_prefix():
         print("Looking for config in user home ~/.local")
         install_prefix = os.path.expanduser('~/.local')
     return install_prefix
+
+# The function below, is taken from
+#
+# https://github.com/Yelp/ephemeral-port-reserve/blob/master/ephemeral_port_reserve.py
+#
+# It is provided under the MIT license
+#
+# The MIT License (MIT)
+#
+# Copyright (c) 2016 Yelp
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+
+def reserve_port(port, ip='localhost'):
```

Comment:
Well a simple solution to our problem would be to bind a socket to a port to see if it's free and then use it to create a server. however I found the above script which not only confirms that the port is free but also "reserves it" in the sense that no other application can use that port for a minute or so. So I used this.

All of this port business is because redis fails silently if it doesn't find a valid port

Reply:
The 'reserving' makes sure that any port that is found free is guaranteed to stay free until connected to by the redis server.

Reply:
Is the reserving in some way faster than letting redis decide? I'm just trying to understand what advantage it gives.

Reply:
The problem with letting redis handle ports is this. in your current config, you have the option `daemonize yes`, and for good reason. The problem with this is that if redis does detect a port collision, it fails silently, with error code 0, this then leads to one of the following situations:

an infinite loop in the while loop testing for the server being active, or, worse, 
a connection to a different, already existing server.

This was the best way I found to completely be sure of avoiding port collisions

Reply:
Ok cool.